code/modules/antagonists/clock_cult/clockwork_turfs.dm

```diff
@@ -0,0 +1,574 @@
+	return
+
+/turf/open/floor/clockwork/Initialize()
+	. = ..()
+	if(uses_overlay)
+		new /obj/effect/temp_visual/ratvar/floor(src)
+		new /obj/effect/temp_visual/ratvar/beam(src)
+		realappearence = new /obj/effect/clockwork/overlay/floor(src)
+		realappearence.linked = src
+
+/turf/open/floor/clockwork/Destroy()
+	if(uses_overlay && realappearence)
+		QDEL_NULL(realappearence)
+	return ..()
+
+/turf/open/floor/clockwork/ReplaceWithLattice()
+	. = ..()
+	for(var/obj/structure/lattice/L in src)
+		L.ratvar_act()
+
+/turf/open/floor/clockwork/try_replace_tile(obj/item/stack/tile/T, mob/user, params)
+	return
+
+/turf/open/floor/clockwork/crowbar_act(mob/living/user, obj/item/I)
+	if(islist(baseturfs))
+		if(type in baseturfs)
+			return TRUE
+	else if(baseturfs == type)
+		return TRUE
+	user.visible_message("<span class='notice'>[user] begins slowly prying up [src]...</span>", "<span class='notice'>You begin painstakingly prying up [src]...</span>")
+	if(I.use_tool(src, user, 70, volume=80))
+		user.visible_message("<span class='notice'>[user] pries up [src]!</span>", "<span class='notice'>You pry up [src]!</span>")
+		make_plating()
+	return TRUE
+
+/turf/open/floor/clockwork/make_plating()
+	if(!dropped_brass)
+		new /obj/item/stack/tile/brass(src)
+		dropped_brass = TRUE
+	if(islist(baseturfs))
+		if(type in baseturfs)
+			return
+	else if(baseturfs == type)
+		return
+	return ..()
+
+/turf/open/floor/clockwork/narsie_act()
+	..()
+	if(istype(src, /turf/open/floor/clockwork)) //if we haven't changed type
+		var/previouscolor = color
+		color = "#960000"
+		animate(src, color = previouscolor, time = 8)
+		addtimer(CALLBACK(src, /atom/proc/update_atom_colour), 8)
+
+/turf/open/floor/clockwork/ratvar_act(force, ignore_mobs)
+	return 0
+
+/turf/open/floor/clockwork/ex_act(severity, target)
+	return
+
+/turf/open/floor/clockwork/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	return
+
+/turf/open/floor/clockwork/reebe
+	name = "cogplate"
+	desc = "Warm brass plating. You can feel it gently vibrating, as if machinery is on the other side."
+	icon_state = "reebe"
+	baseturfs = /turf/open/floor/clockwork/reebe
+	uses_overlay = FALSE
+	planetary_atmos = TRUE
+	var/list/heal_people
+
+/turf/open/floor/clockwork/reebe/Initialize()
+	. = ..()
+	heal_people = list()
+
+/turf/open/floor/clockwork/reebe/Destroy()
+	if(LAZYLEN(heal_people))
+		STOP_PROCESSING(SSprocessing, src)
+	. = ..()
+
+/turf/open/floor/clockwork/reebe/Entered(atom/movable/A)
+	. = ..()
+	var/mob/living/M = A
+	if(istype(M) && is_servant_of_ratvar(M))
+		if(!LAZYLEN(heal_people))
+			START_PROCESSING(SSprocessing, src)
+		heal_people += M
+
+/turf/open/floor/clockwork/reebe/Exited(atom/movable/A, atom/newloc)
+	. = ..()
+	if(A in heal_people)
+		heal_people -= A
+		if(!LAZYLEN(heal_people))
+			STOP_PROCESSING(SSprocessing, src)
+
+/turf/open/floor/clockwork/reebe/process()
+	for(var/mob/living/M in heal_people)
+		if(iscarbon(M))
```

Comment:
Why are you doing a typed for loop _and_ a type check?

Reply:
there is a reason for this, but zesko pinged me with a way to do it better anyway